### PR TITLE
Bug 2109 - Force use of X-Requested-With header on API endpoints

### PR DESCRIPTION
**Pull request by jbuck**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/61f49805d9ab978/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/61f49805d9ab978/output.txt

Total script time: 1.31 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/dc8a7fe8400bd0e/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/dc8a7fe8400bd0e/output.txt

Total script time: 1.28 mins

Server is running on http://pr964.butyr.org/templates/basic/
Run the Test Suite: http://pr964.butyr.org/test/

File: cornfield/lib/filter.js [hunk 1]
```diff
@@ -20,6 +20,15 @@ filters = {
...
       }, 500 );
     }
+  },
+  isXHR: function( req, res, next ) {
+    if ( req.header( 'X-Requested-With' ) === 'XMLHttpRequest' ) {
+      next();
+    } else {
+      res.json({
+        error: 'request is not an XMLHttpRequest'
+      }, 500 );
```
Comment (secretrobotron) [inline cornfield/lib/filter.js]: 500 might not be the best error to report. Maybe we can report something more meaningful.

Comment (jbuck) [inline cornfield/lib/filter.js]: Yeah, I'll use 412 instead since it's a client-originated error

**Pull request merged**